cli/pkg/kctrl/cmd/package/release/helm_openapi_schema_gen.go

```diff
@@ -0,0 +1,362 @@
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "object"},
+					&MapItem{Key: propertiesKey, Value: properties}}}
+			case yaml3.SequenceNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: "array"},
+					&MapItem{Key: defaultKey, Value: "[]"},
+					&MapItem{Key: itemsKey, Value: properties}}}
+			case yaml3.ScalarNode:
+				itemsProperties = &Map{[]*MapItem{
+					&MapItem{Key: typeKey, Value: h.openAPITypeFor(value.Content[0].Tag, value.Content[0].Value)}}}
+			}
+			apiKeys = append(apiKeys, &MapItem{Key: itemsKey, Value: itemsProperties})
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVals})
+		sort.Sort(apiKeys)
+		if key == nil {
+			return &Map{Items: apiKeys}, nil
+		}
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.ScalarNode:
+		var apiKeys openAPIKeys
+		description, isPresent := h.getDescriptionFromNode(key)
+		if isPresent {
+			apiKeys = append(apiKeys, &MapItem{Key: descriptionKey, Value: description})
+		}
+
+		valType := h.openAPITypeFor(value.Tag, value.Value)
+		defaultVal, err := h.getDefaultValue(valType, value.Value)
+		if err != nil {
+			return nil, err
+		}
+		apiKeys = append(apiKeys, &MapItem{Key: typeKey, Value: valType})
+		apiKeys = append(apiKeys, &MapItem{Key: defaultKey, Value: defaultVal})
+		if value.Tag == floatTag {
+			apiKeys = append(apiKeys, &MapItem{Key: formatKey, Value: floatVal})
+		}
+
+		sort.Sort(apiKeys)
+		return &Map{Items: []*MapItem{&MapItem{Key: key.Value, Value: &Map{Items: apiKeys}}}}, nil
+	case yaml3.AliasNode:
+		return h.calculateProperties(key, value.Alias)
+	default:
+		return nil, fmt.Errorf("Unrecognized type %T", value.Kind)
+	}
+	return nil, nil
+}
+
+func (h HelmValuesSchemaGen) getDescriptionFromNode(node *yaml3.Node) (string, bool) {
+	if node == nil || node.HeadComment == "" {
+		return "", false
+	}
+
+	comment := node.HeadComment
+	comment = strings.ReplaceAll(comment, "\n##", "")
+	comment = strings.ReplaceAll(comment, "\n#", "")
+	if strings.HasPrefix(comment, fmt.Sprintf("# %s", node.Value)) {
+		return strings.TrimSpace(strings.TrimPrefix(comment, fmt.Sprintf("# %s", node.Value))), true
+	} else if strings.HasPrefix(comment, "# --") {
+		return strings.TrimSpace(strings.TrimPrefix(comment, "# --")), true
+	} else if strings.HasPrefix(comment, "##") {
+		return strings.TrimSpace(strings.TrimPrefix(comment, "##")), true
+	} else if strings.HasPrefix(comment, "#") {
+		return strings.TrimSpace(strings.TrimPrefix(comment, "#")), true
+	}
+	return "", false
+}
+
+func (h HelmValuesSchemaGen) openAPITypeFor(tag, value string) string {
+	switch tag {
+	case boolTag:
+		return "boolean"
+	case floatTag:
+		return "number"
+	case intTag:
+		return "integer"
+	case nullTag:
+		if value == "null" {
+			return "null"
+		}
+	}
+	return "string"
+
+}
+
+func (h HelmValuesSchemaGen) getDefaultValue(tag, value string) (interface{}, error) {
```

Comment:
`getDefaultValue` is always used in conjunction with ``openAPITypeFor`. It seems we could directly `return strconv.ParseBool(value)` in case of `boolTag` and so on...